src/api/api.ts

```diff
@@ -71,6 +71,12 @@ export function getAllLocations(): Observable<{ uuid: string; display: string }[
   );
 }
 
+export function getEncounterProviders(): Observable<{ uuid: string; display: string }[]> {
+  return openmrsObservableFetch(`/ws/rest/v1/provider?v=custom:(uuid,display)`).pipe(
+    map(({ data }) => data['results']),
+  );
+}
```

Comment:
Are you using this function?

Reply:
I had planned to use it my logic for getting the initial and previous value but that might change.

Reply:
Hi @samuelmale 
I've made some changes.
I haven't introduced a schema adaptor as you had earlier suggested. Realized I would have to first map through the entire schema(pages, sections, questions) to fine tune the schema and then I would have to map through the schema again to process sections. I therefore decided to add this transformation at the section level.
I've introduced the `transformSchema` function which checks every question in a section to see if it needs to be transformed. Let me know what you think.

Reply:
@samuelmale thank  you so much for these fruitful comments which I agree. I think there is a lot of things that need improvement on this code base and perhaps we should add some or most of this type of comments (in case they were not resolved yet) to the concept note (links to the PRs) so that we can work on them when we in Kampala next month for the hackathon. 

@kajambiya I remember that in out tech session today we agreed that this `transformSchema` belongs to the entry-point of the engine where the initial transformations happen. Unless you can't do there because of the additional looping your are talking about then that is fine otherwise it will slowdown the engine.

Reply:
>I remember that in out tech session today we agreed that this transformSchema belongs to the entry-point of the engine where the initial transformations happen.

I agree with you @eudson and the best place I can think of is [here](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/hooks/useFormJson.tsx#L59)

Reply:
Done